### PR TITLE
Bug fix + test : Initializing states for ConvLSTM2D

### PR DESCRIPTION
**Pull request by farizrahman4u**

### CODE REVIEW

**Comment by farizrahman4u** [comment]
@fchollet @xn8812

File: keras/layers/convolutional_recurrent.py [hunk 1]
```diff
@@ -330,9 +331,10 @@ def __init__(self, filters,
         self.recurrent_dropout = min(1., max(0., recurrent_dropout))
 
     def build(self, input_shape):
-        # TODO: better handling of input spec
-        self.input_spec = InputSpec(shape=input_shape)
-
+        if isinstance(input_shape, list):
+            input_shape = input_shape[0]
+        batch_size = input_shape[0] if self.stateful else None
+        self.input_spec[0] = InputSpec(shape=(batch_size,) + input_shape[1:])
```
Comment (fchollet) [inline keras/layers/convolutional_recurrent.py]: `input_spec` contains constraints that future inputs should respect. This statement would set the length of the first input as a length constraint, but unless the network is unrolled there should be no constraint on length. This is in part what the previous TODO was referring to.

**Comment by fchollet** [comment]
In general, having an `InputSpec` instance as `input_spec` should achieve the same as having a list with one element. So I would recommend converting `input_spec` to a list before concatenation in `call`. This will allow users to write custom RNNs without having to set `input_spec` to a list.

**Comment by farizrahman4u** [comment]
Copypasta from Recurrent.py.. so should be fixed there too I guess?

**Comment by fchollet** [comment]
Ideally yes.

**Comment by farizrahman4u** [comment]
Since this is a bug fix, merge it for now and fix the InputSpec thing everywhere in a different PR?

**Comment by fchollet** [comment]
Sure, we can do that.

**Pull request merged**